src/NServiceBus.AcceptanceTests/Sagas/When_a_existing_saga_instance_exists.cs

```diff
@@ -2,47 +2,41 @@
 {
     using System;
     using System.Threading.Tasks;
-    using EndpointTemplates;
-    using AcceptanceTesting;
+    using NServiceBus.AcceptanceTesting;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NServiceBus.Saga;
     using NUnit.Framework;
-    using Saga;
-    using ScenarioDescriptors;
 
     public class When_a_existing_saga_instance_exists : NServiceBusAcceptanceTest
     {
-        static Guid IdThatSagaIsCorrelatedOn = Guid.NewGuid();
-
         [Test]
         public async Task Should_hydrate_and_invoke_the_existing_instance()
         {
-            await Scenario.Define<Context>()
+            var context = await Scenario.Define<Context>()
                 .WithEndpoint<SagaEndpoint>(b => b.Given(bus =>
                     {
-                        bus.SendLocal(new StartSagaMessage { SomeId = IdThatSagaIsCorrelatedOn });
-                        bus.SendLocal(new StartSagaMessage { SomeId = IdThatSagaIsCorrelatedOn, SecondMessage = true });
+                        bus.SendLocal(new StartSagaMessage { SomeId = Guid.NewGuid() });
                         return Task.FromResult(0);
                     }))
                 .Done(c => c.SecondMessageReceived)
-                .Repeat(r => r.For(Persistence.Default))
-                .Should(c => Assert.AreEqual(c.FirstSagaInstance, c.SecondSagaInstance, "The same saga instance should be invoked invoked for both messages"))
                 .Run();
+
+            Assert.AreEqual(context.FirstSagaId, context.SecondSagaId, "The same saga instance should be invoked invoked for both messages");
         }
 
         public class Context : ScenarioContext
         {
             public bool SecondMessageReceived { get; set; }
 
-            public Guid FirstSagaInstance { get; set; }
-            public Guid SecondSagaInstance { get; set; }
+            public Guid FirstSagaId { get; set; }
+            public Guid SecondSagaId { get; set; }
         }
 
         public class SagaEndpoint : EndpointConfigurationBuilder
         {
             public SagaEndpoint()
             {
-                EndpointSetup<DefaultServer>(
```

Comment:
Think so, when you remove the wrapping the sends will "escape" the transaction and fail the test